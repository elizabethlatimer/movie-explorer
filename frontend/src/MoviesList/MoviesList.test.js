import React from 'react';
import { render } from "@testing-library/react";
import MovieList from './index';
import MovieContext from '../shared/helpers/movieContext'
import { BrowserRouter, Route } from 'react-router-dom';

const values = {
  movieList: {
    "movies": {
      "numResults": 150,
      "results": [
        {
          "id": 618354,
          "title": "Superman: Man of Tomorrow",
          "overview": "It’s the dawn of a new age of heroes, and Metropolis has just met its first. But as Daily Planet intern Clark Kent – working alongside reporter Lois Lane – secretly wields his alien powers of flight, super-strength and x-ray vision in the battle for good, there’s even greater trouble on the horizon.",
          "thumbnailPoster": "https://image.tmdb.org/t/p/w185/6Bbq8qQWpoApLZYWFFAuZ1r2gFw.jpg",
          "mainPoster": "https://image.tmdb.org/t/p/w342/6Bbq8qQWpoApLZYWFFAuZ1r2gFw.jpg",
          "releaseDate": "August 22, 2020",
          "director": "Chris Palmer"
        }]
    }
  },
  currentQuery: "superman",
  search: () => { }
};

test('renders loading without crashing', () => {
  render(
    <MovieContext.Provider value={{ ...values, movieList: null }}>
      <BrowserRouter>
        <MovieList />
      </BrowserRouter>
    </MovieContext.Provider>);
});

test('matches the loading snapshot', () => {
  const { asFragment } = render(
    <MovieContext.Provider value={{ ...values, movieList: null }}>
      <BrowserRouter>
        <MovieList />
      </BrowserRouter>
    </MovieContext.Provider>);
  expect(asFragment()).toMatchSnapshot();
});

test('renders results without crashing', () => {
  render(
    <MovieContext.Provider value={values}>
      <BrowserRouter>
        <MovieList />
      </BrowserRouter>
    </MovieContext.Provider>);
});

test('matches the results snapshot', () => {
  const { asFragment } = render(
    <MovieContext.Provider value={values}>
      <BrowserRouter>
        <MovieList />
      </BrowserRouter>
    </MovieContext.Provider>);
  expect(asFragment()).toMatchSnapshot();
});