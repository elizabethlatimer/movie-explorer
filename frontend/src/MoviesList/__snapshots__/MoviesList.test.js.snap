// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the loading snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner-border"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</DocumentFragment>
`;

exports[`matches the results snapshot 1`] = `
<DocumentFragment>
  <div
    class="MoviesList"
  >
    <h2>
      Search Results
    </h2>
    <div
      class="MovieListCard"
    >
      <a
        href="/movies/618354"
      >
        <div
          class="card"
        >
          <div
            class="card-horizontal"
          >
            <div
              class="img-square-wrapper"
            >
              <img
                alt="Superman: Man of Tomorrow poster"
                class=""
                src="https://image.tmdb.org/t/p/w185/6Bbq8qQWpoApLZYWFFAuZ1r2gFw.jpg"
              />
            </div>
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                Superman: Man of Tomorrow
              </div>
              <p
                class="card-text"
              >
                It’s the dawn of a new age of heroes, and Metropolis has just met its first. But as Daily Planet intern Clark Kent – working alongside reporter Lois Lane – secretly wields his alien powers of flight, super-strength and x-ray vision in the battle for good, there’s even greater trouble on the horizon.
              </p>
              <p
                class="card-text"
              >
                Release date: August 22, 2020
              </p>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
